File: app/controllers/api/events_controller.rb
module Api
  class EventsController < ApplicationController
    respond_to :json

    def index
      respond_with Event.all
    end

    def search
      query = params[:query]
      events = Event.where('name Like ? OR place LIKE ? OR description LIKE ?', "%#{query}%", "%#{query}%", "%#{query}%")

      respond_with evetns
    end

    def create
      event = Event.new(event_params)
      if event.save
        respond_with :api, event, status: :ok, location: api_events_url
      else
        render json: { errors: event.errors.full_messages }, status: :unprocessable_entity
      end
    end

    def update

    end

    def destroy

    end

    private

    def event_params
      params.requrie(:event).permit(:name, :description, :event_date, :place)
    end
  end
end

File: app/assets/javascripts/angular/events/EventsController.js
var app = angular.module('railsAngular');

app.controller('EventsController', ['$scope', function($scope, Event) {
  $scope.events = Event.query();

  $scope.addEvent = function() {
    if (!valid()) {
      return false;
    }

    Event.save($scope.event,
      function(response, _headers) {
        $scope.events.push(response);
      },
      function(response) {
        alert('Errors: ' + response.date.errors.join(. ));
      }
    );

    $scope.event = {};
  };

  valid = function() {
    return !!$scope.event &&
      !!$scope.event.name && !!$scope.event.event_date &&
      !!$scope.event.description && !!$scope.event.place;
  }
}]);